### DISCUSSION TITLE
JanusTextRoomPlugin When I try to attach, and TCP/IP add a lot of connections

### DISCUSSION THREAD

**Discussion by alantin918** [flutterjanus/flutter_janus_client#80] [Q&A] [answered]
Hello~

    rest = RestJanusTransport(url: servermap['janus_rest']);
    janusClient = JanusClient(
        withCredentials: true,
        apiSecret: "SecureIt",
        transport: rest,
        iceServers: [
          RTCIceServer(
              urls: "stun:stun1.l.google.com:19302",
              username: userNickname.text,
              credential: "")
        ]);

    session = await janusClient.createSession();
    textRoom = await session.attach<JanusTextRoomPlugin>();

<img width="928" alt="截圖 2022-08-18 上午11 06 16" src="https://user-images.githubusercontent.com/52562028/185284308-ef54b2e7-ff71-44ac-8f5b-d6aa3d157f5a.png">

When I try to attach session,
my netstat increases the number of connections a lot.
I don't know why...

**Comment by shivanshtalwar** [comment, marked as answer]
That is because, RestTransport works by polling that is it continuously sends request to janus rest endpoint at certain fixed interval in order to get events from janus

**Reply by alantin918** [reply]
So RestTransport 
the number of connections is normal？

The connection is maintained 30 times

<img width="922" alt="截圖 2022-08-18 下午2 32 12" src="https://user-images.githubusercontent.com/52562028/185310191-81b766ce-8e68-4c22-8d4c-d02c486d7956.png">

**Reply by shivanshtalwar0** [reply]
Yes it is normal if you feel that is too much you can stick with WebsocketTransport it will only maintain one connection
